### PR TITLE
Frontend typed subscriptions

### PR DESCRIPTION
**Pull request by thisconnect**
The plan was to look at all subscriptions, but I noticed some funny code in accountsummary that needed more refactoring before changing to syncdone and statusChanged typed subscriptions.

I'll leave some comments in the PR, so that it is easier to review.

Test with

    rm -rf appfolder.dev/cache/account-* && make servewallet

### CODE REVIEW

File: frontends/web/src/api/accountsync.ts [hunk 1]
```diff
@@ -43,3 +45,35 @@ export const syncAddressesCount = (
...
+export const syncdone = (
+  code: string,
+  cb: (code: string) => void,
+): TUnsubscribe => {
+  return subscibeLegacy('syncdone', data => {
+    if (data.type === 'account' && data.code === code) {
+      cb(code);
+    }
+  });
+};
```
Comment (thisconnect) [inline frontends/web/src/api/accountsync.ts:79]: statusChanged and syncdone was just moved from api/subscribe-legacy.ts, as it should be in api/accountsync.

File: frontends/web/src/routes/account/send/send.tsx [hunk 2]
```diff
@@ -46,10 +47,11 @@ import { FeeTargets } from './feetargets';
 import style from './send.module.css';
 import { TSelectedUTXOs, UTXOs } from './utxos';
 import { route } from '../../../utils/route';
+import { UnsubscribeList, unsubscribe } from '../../../utils/subscriptions';
 
 interface SendProps {
     accounts: accountApi.IAccount[];
-    code?: string;
+    code: string;
```
Comment (thisconnect) [inline frontends/web/src/routes/account/send/send.tsx:54]: changed to making code prop mandatory, so we don't have to do `if (this.props.code) { ` in the subscription callback

File: frontends/web/src/routes/account/send/send.tsx [hunk 3]
```diff
@@ -161,34 +163,31 @@ class Send extends Component<Props, State> {
...
+          const { data, meta, type } = payload;
+          switch (type) {
+          case 'device':
+            switch (data) {
+            case 'signProgress':
+              this.setState({ signProgress: meta, signConfirm: false });
+              break;
+            case 'signConfirm':
+              this.setState({ signConfirm: true });
+              break;
```
Comment (thisconnect) [inline frontends/web/src/routes/account/send/send.tsx:179]: I'll add typed subscriptions for devicesync.signProgress and devicesync.signConfrim in a later commit or PR.

File: frontends/web/src/routes/account/summary/accountssummary.tsx [hunk 4]
```diff
@@ -45,7 +47,7 @@ export interface Balances {
 }
 
 interface SyncStatus {
-    [code: string]: string;
+    [code: string]: number;
 }
```
Comment (thisconnect) [inline frontends/web/src/routes/account/summary/accountssummary.tsx:51]: Better to have typed subscriptions.

File: frontends/web/src/routes/account/summary/accountssummary.tsx [hunk 5]
```diff
@@ -183,7 +193,7 @@ class AccountsSummary extends Component<Props, State> {
     if (initial) {
       return;
     }
-    return await this.getAccountsTotalBalance();
+    this.getAccountsTotalBalance();
```
Comment (thisconnect) [inline frontends/web/src/routes/account/summary/accountssummary.tsx:196]: Also removed unnecessary return await here

**Comment by thisconnect** [comment]
rebased and added a commit to rename argument to `event` (was `data`)

**Review by Beerosagos — APPROVED** [review]
LGTM! Very nice :pray:

File: frontends/web/src/api/accountsync.ts [hunk 6]
```diff
@@ -14,7 +14,9 @@
  * limitations under the License.
  */
 
-import { subscribeEndpoint, TUnsubscribe } from './subscribe';
+import { TUnsubscribe } from '../utils/transport-common';
+import { subscribeEndpoint } from './subscribe';
+import { subscribe as subscibeLegacy } from '../utils/event-legacy';
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/api/accountsync.ts]: fixed

File: frontends/web/src/routes/account/summary/accountssummary.tsx
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/routes/account/summary/accountssummary.tsx:1]: done

**Pull request merged**